app/assets/javascripts/analytics/_events.js

```diff
@@ -97,8 +97,31 @@
         });
       }
     },
+    'awardDetailsSubmitButtonClick': function () {
+      var currentURL = GOVUK.GDM.analytics.location.pathname();
+
+      // check we're on the award details page
+      if (currentURL.match('\/buyers\/frameworks\/[a-z0-9-]+\/requirements\/[a-z0-9-]+\/[0-9]+\/award\/[0-9]+\/contract-details')) {
+        GOVUK.analytics.trackEvent('Buyer Award', 'Awarded');
+      }
+
+    },
+    'cancelAwardSubmitButtonClick': function () {
+      var currentURL = GOVUK.GDM.analytics.location.pathname();
+
+      // check we're on the 'why did you cancel' page
+      if (currentURL.match('\/buyers\/frameworks\/[a-z0-9-]+\/requirements\/[a-z0-9-]+\/[0-9]+\/cancel-award')) {
+        var selectedCancelReason = $("input[name='cancel_reason']:checked").val();
+        if (selectedCancelReason) {
+          GOVUK.analytics.trackEvent('Buyer Award', 'Not-awarded', {'label': selectedCancelReason});
+        }
+      }
+
+    },
     'init': function () {
       $('body').on('click', 'a', this.supplierListDownload)
+        .on('click', 'input[type=submit]', this.awardDetailsSubmitButtonClick)
+        .on('click', 'input[type=submit]', this.cancelAwardSubmitButtonClick);
```

Comment:
Seems a bit weird to run two functions here, maybe we could merge them into one _awardOrCancel_ function and just have something like 
```
var currentURL = GOVUK.GDM.analytics.location.pathname();
var awardBriefURL = '\/buyers\/frameworks\/[a-z0-9-]+\/requirements\/[a-z0-9-]+\/[0-9]+\/award\/[0-9]+\/contract-details';
var cancelBriefURL = '\/buyers\/frameworks\/[a-z0-9-]+\/requirements\/[a-z0-9-]+\/[0-9]+\/cancel-award';


if (currentURL.match(awardBriefUrl)) {
  GOVUK.analytics.trackEvent('Buyer Award', 'Awarded');
}
else if (currentURL.match(cancelBriefURL)) {
  var selectedCancelReason = $("input[name='cancel_reason']:checked").val();
  if (selectedCancelReason) {
    GOVUK.analytics.trackEvent('Buyer Award', 'Not-awarded', {'label': selectedCancelReason});
  }
}
```
It's not a big deal having two functions anyway, so whatever is easier for you.

Reply:
Ah I think @katstevens and me looked at this before and settled on separate functions because they're supposed to be triggered by separate buttons. I asked the same thing though 😀